--- a/src/certificate_authority/rcgen_authority.rs
+++ b/src/certificate_authority/rcgen_authority.rs
@@ -3,7 +3,8 @@
 use moka::future::Cache;
 use rand::{thread_rng, Rng};
 use rcgen::{
-    Certificate, CertificateParams, DistinguishedName, DnType, Ia5String, KeyPair, SanType,
+    Certificate, CertificateParams, DistinguishedName, DnType, ExtendedKeyUsagePurpose, Ia5String,
+    KeyPair, KeyUsagePurpose, SanType,
 };
 use std::sync::Arc;
 use time::{Duration, OffsetDateTime};
@@ -75,6 +76,10 @@
             Ia5String::try_from(authority.host()).expect("Failed to create Ia5String"),
         ));
 
+        params.key_usages = vec![KeyUsagePurpose::DigitalSignature];
+        params.extended_key_usages = vec![ExtendedKeyUsagePurpose::ServerAuth];
+        params.use_authority_key_identifier_extension = true;
+
         params
             .signed_by(&self.key_pair, &self.ca_cert, &self.key_pair)
             .expect("Failed to sign certificate")
